### PR TITLE
9406 star rating UI and JSON parsing

### PR DESCRIPTION
**Pull request by adam-govan**
Added Star Rating Conversation and refactored Atom Factory to allow Compound objects

Also moved GCC workaround to SwrveSetup

WIP on the UI Stars (needs more polish)

@Sergio-Mira / @dominicmarmionswrve

### CODE REVIEW

File: Sdk/Conversation/SwrveConversationStyler.m [hunk 1]
```diff
@@ -95,4 +95,17 @@ +(void) styleButton:(SwrveConversationUIButton*)button withStyle:(NSDictionary*)
...
+
+    NSString *styleType = kSwrveTypeSolid;
+    if([style objectForKey:kSwrveKeyType]) {
+        styleType = [style objectForKey:kSwrveKeyType];
+    }
+    NSString *bgHexColor = [self colorFromStyle:[style objectForKey:kSwrveKeyBg] withDefault:kSwrveDefaultColorBg];
+    UIColor *bgUIColor = [self convertToUIColor:bgHexColor];
+    UIColor *starColor = [self convertToUIColor:starColorHex];
+    
+    [ratingView initRatingTypewithStarColor:starColor withBackgroundColor:bgUIColor];
```
Comment (Sergio-Mira) [inline Sdk/Conversation/SwrveConversationStyler.m]: initRatingTypeWithStarColor

File: Sdk/Conversation/SwrveConversationAtomFactory.m [hunk 2]
```diff
@@ -49,18 +53,26 @@ +(SwrveConversationAtom *) atomForDictionary:(NSDictionary *)dict {
...
-        return swrveInputMultiValue;
+        [atomArray addObject:swrveInputMultiValue];
+        
     } else if ([type isEqualToString:kSwrveContentSpacer]) {
         SwrveContentSpacer* swrveContentSpacer = [[SwrveContentSpacer alloc] initWithTag:tag andDictionary:dict];
         swrveContentSpacer.style = [dict objectForKey:@"style"];
-        return swrveContentSpacer;
+        [atomArray addObject:swrveContentSpacer];
+    } else if ([type isEqualToString:kSwrveControlStarRating]) {
+        SwrveContentHTML *swrveContentHTML = [[SwrveContentHTML alloc] initWithTag:tag andDictionary:dict];
```
Comment (Sergio-Mira) [inline Sdk/Conversation/SwrveConversationAtomFactory.m:67]: :+1:

File: Sdk/Conversation/SwrveConversationStarRatingView.m [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        UIImageView *imageView = [[UIImageView alloc] init];
+        imageView.contentMode = UIViewContentModeScaleAspectFit;
+        [self.swrveStarViews addObject:imageView];
+        [self addSubview:imageView];
+    }
+    [self setNeedsLayout];
+    [self refresh];
+}
+
+- (void) initRatingTypewithStarColor:(UIColor *) starColor withBackgroundColor:(UIColor *)backgroundColor
```
Comment (Sergio-Mira) [inline Sdk/Conversation/SwrveConversationStarRatingView.m]: Can we change the name to other than init*? It does not call [super init](and it shouldn't)

File: Sdk/Conversation/SwrveConversationStarRatingView.m [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    CGFloat imageWidth = SWRVEMAX(self.swrveMinImageSize.width, desiredImageWidth);
+    CGFloat imageHeight = SWRVEMAX(self.swrveMinImageSize.height, self.frame.size.height);
+    for (NSUInteger i = 0; i < self.swrveStarViews.count; ++i) {
+        UIImageView *imageView = [self.swrveStarViews objectAtIndex:i];
+        CGRect imageFrame = CGRectMake((self.swrveLeftMargin + i) * (self.swrveMidMargin+imageWidth), 0, imageWidth, imageHeight);
+        imageView.frame = imageFrame;
+    }
+}
+
+#pragma mark - iOS6 change tint support
```
Comment (Sergio-Mira) [inline Sdk/Conversation/SwrveConversationStarRatingView.m:119]: :+1:

**Comment by Sergio-Mira** [comment]
:rocket:

File: Sdk/Conversation/SwrveConversationStarRatingView.m (hunk 3 above)
Comment (adam-govan) [inline Sdk/Conversation/SwrveConversationStarRatingView.m]: that's a good point, maybe something like: updateRatingTypeWithStarColor?

Comment (Sergio-Mira) [inline Sdk/Conversation/SwrveConversationStarRatingView.m]: sounds good!

File: Sdk/Conversation/SwrveConversationStarRating.h [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#import "SwrveConversationAtom.h"
+#import "SwrveConversationStarRatingView.h"
+
+@interface SwrveConversationStarRating : SwrveConversationAtom <SwrveConversationStarRatingViewDelegate>
+
+
```
Comment (dominicmarmionswrve) [inline Sdk/Conversation/SwrveConversationStarRating.h]: extra line

File: Sdk/Conversation/SwrveConversationStarRatingView.m [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+              [self setNeedsLayout];
+              [self refresh];
+}
+
+- (void) refresh {
+    for(NSUInteger i = 0; i < self.swrveStarViews.count; ++i) {
+        UIImageView *imageView = [self.swrveStarViews objectAtIndex:i];
+        if (self.swrveCurrentRating >= i+1) {
+            imageView.image = self.swrveFullSelectedImage;
+            if([imageView respondsToSelector:@selector(setTintColor:)]){
```
Comment (dominicmarmionswrve) [inline Sdk/Conversation/SwrveConversationStarRatingView.m:85]: :+1:

**Comment by dominicmarmionswrve** [comment]
:rocket:

**Pull request merged**